### PR TITLE
Add `hamronization/summarize`

### LINKED ISSUE #1271: new module: hamronization
I think it would be good to have a module for [hamronization](https://bioconda.github.io/recipes/hamronization/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [ ] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Second step of hAMRonization (gathering all the tool-specific command outputs)

Closes #1271 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/hamronization/summarize/meta.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      type: file
+      description: hAMRonised summary in JSON format
+      pattern: "*.json"
+  - tsv:
+      type: file
+      description: hAMRonised summary in TSV format
+      pattern: "*.json"
+  - html:
+      type: file
+      description: hAMRonised summary in html format
```
Comment (jasmezz) [inline modules/hamronization/summarize/meta.yml:41]:
```suggestion
      description: hAMRonised summary in HTML format
```

**Review by jasmezz — APPROVED** [review]

File: modules/hamronization/summarize/main.nf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def outformat = format == 'interactive' ? 'html' : format
+    """
+    hamronize \\
+        summarize \\
+        ${reports.join(' ')} \\
```
Comment (rpetit3) [inline modules/hamronization/summarize/main.nf:28]: Does `$reports` (without join) work here? (honestly just curious!)

**Review by rpetit3 — APPROVED** [review]

File: modules/hamronization/summarize/main.nf (hunk 2 above)
Comment (jfy133) [reply modules/hamronization/summarize/main.nf:28]: Actually it does! I'll leave it like this as it works and I have two approvals but we can always remove it in the future if it's unnecesary or cause problems :+1:

Comment (rpetit3) [reply modules/hamronization/summarize/main.nf:28]: Awesome! Thanks for checking, can add it to the "Today I learned" book

**Pull request merged**